realm/realm-library/src/main/cpp/util.hpp

```diff
@@ -572,6 +574,10 @@ class JniByteArray {
         , m_arrayLength(javaArray == NULL ? 0 : env->GetArrayLength(javaArray))
         , m_array(javaArray == NULL ? NULL : env->GetByteArrayElements(javaArray, NULL))
         , m_releaseMode(JNI_ABORT) {
+        if (m_javaArray != nullptr && m_array == nullptr) {
+            // javaArray is not null but GetByteArrayElements returns null, something is really wrong.
```

Comment:
is the data copied in the case of `GetByteArrayElements` returning null? (you can find out using the flag `idCopy`)